src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -47,6 +49,7 @@ public class ModelMetadata implements Writeable, ToXContentObject {
     final private String timestamp;
     final private String description;
     private String error;
+    private String nodeAssignment;
```

Comment:
I didn't set it as final, because when the job is created it doesn't initially have an assignment.  The assignment only happens after the creation of the object.